src/test/jetpack-editor-setup.js

```diff
@@ -47,15 +47,26 @@ const jetpackEmbedVariations = [
 	'smartframe',
 ];
 
+/**
+ * Registers the Jetpack blocks using a sandbox registry for the modules via `jest.jest.isolateModules`.
+ * This is required for the Jetpack blocks because they are registered upon module importation.
+ *
+ * @param {Object} props
+ * @return {Object} Blocks API registered in the sandbox registry.
+ */
+const registerJetpackBlocksIsolated = ( props ) => {
+	let blocksAPI;
+	jest.isolateModules( () => {
+		registerJetpackBlocks( props );
+		blocksAPI = require( '@wordpress/blocks' );
+	} );
+	return blocksAPI;
+};
+
 describe( 'Jetpack blocks', () => {
 	afterEach( () => {
 		// Reset Jetpack data
 		delete global.window[ JETPACK_DATA_PATH ];
-
-		// Clean up registered blocks
-		getBlockTypes().forEach( ( block ) => {
-			unregisterBlockType( block.name );
-		} );
```

Comment:
This is no longer needed because when using `registerJetpackBlocksIsolated` function, a new block registry is instanced in each call for registering the Jetpack blocks.